wgsl/index.bs

```diff
@@ -9397,11 +9431,20 @@ Each structure member in the entry point IO [=shader-creation error|must=] be on
 (see [[#builtin-inputs-outputs]]), or assigned a location.
 
 Locations [=shader-creation error|must not=] overlap within each of the following sets:
-* Members within a structure type.
+* Members within a structure type for a given [=attribute/blend_src=] value (a single set if none is specified).
     This applies to any structure, not just those used in shader stage inputs or outputs.
 * An entry point's shader stage inputs,
     i.e. locations for its formal parameters, or for the members of its formal parameters of structure type.
 
+When the [=attribute/blend_src=] attribute is specified:
```

Comment:
It seems ambiguous what set of things we're looking at for this validation. The one above says "within each of the following sets" but that doesn't clearly carry through to this paragraph. I do not know how to word this well.

Reply:
In my latest commit I only apply the “location overlapping" rules on the members without `blend_src`, and for the members with `blend_src` there are special and much stricter rules for them below `when the [=attribute/blend_src=] attribute is specified`:
- `blend_src` can only be applied on fragment outputs.
- only two outputs are allowed.
- `location` must always be `0`.

What do you think?

Reply:
It's still unclear what sets these apply to. Maybe we want something like this (but formalized a little to link to the right things and use the right language - I'm not too familiar with this spec). Does it seem right?

- For each set of outputs |outputs| in the following sets:
  - members within a structure type (...)
  - an entry point's shader stage inputs (...)

  The following must be true:

  - |outputs| must not contain two outputs with the same location and no blend_src.
  - If any output in |outputs| specifies a blend_src:
    - |outputs| must contain exactly 2 outputs, one with `@location(0) @blend_src(0)` and one with `@location(0) @blend_src(1)`

Reply:
What do you think? @alan-baker

Reply:
@kainino0x  Done. PTAL, thanks!